File: 3354.make-array-elements-equal-to-zero/solution.go
func countValidSelections(nums []int) int {
    res := 0
    total := 0
    for _, n := range nums {
        total += n
    }
    prefixSum := 0
    for _, n := range nums {
        prefixSum += n
        if n == 0 {
            d := total - 2 * prefixSum
            if d == 0 {
                res += 2
            } else if d == 1 || d == -1 {
                res++
            }
        }
    }
    return res
}

File: 3354.make-array-elements-equal-to-zero/solution.kt
class Solution {
    fun countValidSelections(nums: IntArray): Int {
        val total = nums.sum()
        var res = 0
        var prefix = 0

        nums.forEach { n ->
            prefix += n
            if (n == 0) {
                res += when(total - 2 * prefix) {
                    0     -> 2
                    -1, 1 -> 1
                    else  -> 0
                }
            }
        }

        return res
    }
}